src/figuresClasses.ts

```diff
@@ -1,19 +1,80 @@
-export interface Figure {
 
-}
+type Shape = 'triangle' | 'circle' | 'rectangle';
 
-export class Triangle {
+type Color = 'red' | 'green' | 'blue';
 
+export interface Figure {
+  shape: Shape,
+  color: Color,
+  getArea(): number,
 }
 
-export class Circle {
+export class Triangle implements Figure {
+  shape: Shape = 'triangle';
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a === 0 || b === 0 || c === 0) {
+      throw new Error('One of the sides is Zero(0)');
+    }
+
+    if (a + b <= c || c + b <= a || a + c <= b) {
+      throw new Error('Use proper values to calculate area');
+    }
+  }
+
+  getArea(): number {
+    const p: number = 0.5 * (this.a + this.b + this.c);
+    const S: number = Math.sqrt(p * ((p - this.a)
```

Comment:
What is `p`, `S` rename these variables
```suggestion
    const p = 0.5 * (this.a + this.b + this.c);
    const S = Math.sqrt(p * ((p - this.a)
```